kiva/pyagg.py

```diff
@@ -0,0 +1,725 @@
+                        units='userSpaceOnUse'):
+        """ Sets a linear gradient as the current brush.
+        """
+        spread, units = self._get_gradient_enums(spread_method, units)
+        self.fill_paint = agg.LinearGradientPaint(x1, y1, x2, y2,
+                                                  stops, spread, units)
+
+    def radial_gradient(self, cx, cy, r, fx, fy, stops, spread_method,
+                        units='userSpaceOnUse'):
+        """ Sets a radial gradient as the current brush.
+        """
+        spread, units = self._get_gradient_enums(spread_method, units)
+        self.fill_paint = agg.RadialGradientPaint(cx, cy, r, fx, fy,
+                                                  stops, spread, units)
+
+    # ----------------------------------------------------------------
+    # Drawing Images
+    # ----------------------------------------------------------------
+
+    def draw_image(self, img, rect=None):
+        """
+        img is either a N*M*3 or N*M*4 numpy array, or a Kiva image
+
+        rect - a tuple (x, y, w, h)
+        """
+        if isinstance(img, np.ndarray):
+            # Numeric array
+            if img.shape[2] == 3:
+                format = agg.PixelFormat.RGB24
+            elif img.shape[2] == 4:
+                format = agg.PixelFormat.RGBA24
+            draw_img = agg.Image(img.astype(np.uint8), format)
+        elif isinstance(img, agg.CanvasRGB24):
+            draw_img = img.image
+        elif isinstance(img, GraphicsContext):
+            # An offscreen kiva context
+            draw_img = img.gc.image
+        else:
+            msg = "Cannot render image of type '{}' into pyagg context."
+            warnings.warn(msg.format(type(img)))
+            return
+
+        transform = agg.Transform()
+        transform.multiply(self.transform)
+        transform.translate(*rect[:2])  # XXX: Need to use w & h too!
+
+        self.gc.draw_image(draw_img.array, draw_img.pixel_format, transform,
+                           self.canvas_state)
+
+    # ----------------------------------------------------------------
+    # Drawing Text
+    # ----------------------------------------------------------------
+
+    def select_font(self, name, size, textEncoding):
+        """ Set the font for the current graphics context.
+        """
+        self.font = agg.Font(name, size, agg.FontCacheType.VectorFontCache)
+
+    def set_font(self, font):
+        """ Set the font for the current graphics context.
+        """
+        self.select_font(font.findfont(), font.size, None)
+
+    def set_font_size(self, size):
+        msg = "set_font_size not implemented on pyagg yet."
+        raise NotImplementedError(msg)
+
+    def set_character_spacing(self, spacing):
+        msg = "set_character_spacing not implemented on pyagg yet."
+        raise NotImplementedError(msg)
+
+    def set_text_drawing_mode(self):
+        """ XXX: What is this for??
```

Comment:
Yes, if pyagg has vector font support that should cover it if you can specify the render method (fill, stroke, etc.) for the vector path of the text.